### PR TITLE
Run3-alca245 Modify the muon analysis code for calibrating inter-depths in HB/HE

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Modify the muon analysis code for calibrating inter-depths in HB/HE

#### PR validation:

Tested by running the script in Calibration/HcalCalibAlgos/test/python

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc [hunk 1]
```diff
@@ -67,7 +69,7 @@
 #include "MagneticField/Engine/interface/MagneticField.h"
 #include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (francescobrivio) [inline Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc:72]: Is this supposed to remain uncommented?

**Comment by francescobrivio** [comment]
@cmsbuild please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc]: Taken care of

File: Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc [hunk 2]
```diff
@@ -1188,12 +1221,33 @@ double HcalHBHEMuonAnalyzer::activeLength(const DetId& id_) {
     dpths.emplace_back(id.depth());
   }
   double lx(0);
+  if (debug) {
+#ifdef EDM_ML_DEBUG
```
Comment (francescobrivio) [inline Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc:1225]: Thanks @bsunanda for addressing my previous comment!
Looking again, I'm now a bit confused by this:
```
if(debug) {
#ifdef EDM_ML_DEBUG
...
}
```
What is the purpose of having two "debug safeguards" here?
Sorry for noticing this only now (and if the answer is obvious)....

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@francescobrivio Please approve this now

**Comment by francescobrivio** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**